src/engine.rs

```diff
@@ -350,13 +352,47 @@ where
     }
 
     /// Dumps all operations.
-    #[allow(unused_variables)]
     pub fn dump_with_file_builder(
         path: &Path,
         raft_groups: &[u64],
         file_builder: Arc<B>,
     ) -> Result<Vec<LogItem>> {
-        todo!()
+        Self::dump_log(path, raft_groups, file_builder)
+    }
+
+    pub fn dump_log(
+        path: &std::path::Path,
+        raft_groups: &[u64],
+        file_builder: Arc<B>,
+    ) -> Result<Vec<LogItem>> {
+        if !path.exists() {
+            return Err(Error::InvalidArgument(format!(
+                "raft-engine directory or file '{}' does not exist.",
+                path.to_str().unwrap()
+            )));
+        }
+
+        let mut reader = if path.is_dir() {
+            LogItemReader::new_directory_reader(file_builder, path).unwrap()
+        } else {
+            LogItemReader::new_file_reader(file_builder, path).unwrap()
+        };
+
+        let mut log_map = HashMap::<u64, Vec<LogItem>>::default();
```

Comment:
No, I mean return an iterator to user, instead of reading all the items into vector.

Reply:
Return an iterator to user instead of a vector is for the possibility that data may be too large and we can afford to allocate so many memory? @tabokie

Reply:
Yes, and user might not need all of the items.

Reply:
Got it